### PR TITLE
added functionality to filter component 

### PR DESCRIPTION
**Pull request by Priyanshi-Tiwari**
- added functionality to filter component
       -sort by price range
       -sort by rating
       -sort by categories
       -sort by price high to low/ low to high

- updated product-listing page

https://user-images.githubusercontent.com/71331989/160833485-76fb5c8d-7718-47a6-9c30-1a0e54356233.mp4

### CODE REVIEW

File: src/contexts/product-data-context.jsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+
+const ProductProvider=({children})=>{
+    const[data,setData]=useState([])
+    console.log("state",data)
+
+    useEffect(()=>{
+    const fetch=async()=>{
+    const result= await axios("/api/products");
+    console.log("result",result)
```
Comment (therajatg) [inline src/contexts/product-data-context.jsx:15]: can remove console.log

File: src/contexts/product-data-context.jsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+const ProductContext=createContext();
+
+
+
+const ProductProvider=({children})=>{
+    const[data,setData]=useState([])
+    console.log("state",data)
+
+    useEffect(()=>{
+    const fetch=async()=>{
```
Comment (therajatg) [inline src/contexts/product-data-context.jsx:13]: have try and catch inside async-awair

File: src/contexts/product-data-context.jsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    return(
+        <ProductContext.Provider value={{data}}>
+            {children}
+        </ProductContext.Provider>
+    )
+}
+
+const useProduct=()=>{
+    return useContext(ProductContext)
+}
```
Comment (therajatg) [inline src/contexts/product-data-context.jsx:29]: can simply write this:

const useProduct=()=>useContext(ProductContext)

**Review by therajatg — APPROVED** [review]
Everything else looks good

File: src/pages/product-listing/product-listing.jsx [hunk 4]
```diff
@@ -2,15 +2,28 @@ import React from 'react'
 import './product-listing.css'
 import { Card } from '../../components/card/card'
 import { Filter } from '../../components/filter/filter'
+import { useProduct } from '../../contexts/product-data-context'
+import { useFilter } from '../../contexts/filter-context'
+import {sortedData, filteredData} from './utils'
 
 const ProductListing= () => {
+  const{state}=useFilter();
+  console.log("filterstate",state)
```
Comment (sadathshariff) [inline src/pages/product-listing/product-listing.jsx:11]: remove console logs

**Review by sadathshariff — APPROVED** [review]
Overall the code looks good

File: src/contexts/product-data-context.jsx (hunk 1 above)
Comment (Priyanshi-Tiwari) [reply src/contexts/product-data-context.jsx:15]: will do it

File: src/pages/product-listing/product-listing.jsx (hunk 4 above)
Comment (Priyanshi-Tiwari) [reply src/pages/product-listing/product-listing.jsx:11]: will do it

**Pull request merged**